kratos/python_scripts/restart_utility.py

```diff
@@ -175,6 +200,48 @@ def CreateOutputFolder(self):
                 os.makedirs(folder_path)
             self.model_part.GetCommunicator().GetDataCommunicator().Barrier()
 
+    def GetRestartFiles(self):
+        """Return a dictionary of stepID - restart_file_list dictionary that stores sets of restart files for each step."""
+        restart_path    = Path( self.__GetFolderPathSave() )
+        restart_files   = {}
+        if restart_path.is_dir():
+
+            file_name_data_collector = KratosMultiphysics.FileNameDataCollector( self.model_part, self._GetFileNamePattern(), {} )
+            file_name_data_list = file_name_data_collector.GetFileNameDataList()
```

Comment:
For OpenMP this works, for MPI for whatever reason it tends to return an empty list. This means, after restart, the existing files will not be removed correctly (meaning at all).

Reply:
I'll also take a look :)